ocaml/xapi/xapi_xenops.ml

```diff
@@ -1956,18 +1955,19 @@ let update_vusb ~__context (id: (string * string)) =
         else begin
           let pusb, pusb_r =
             Db.VM.get_VUSBs ~__context ~self:vm
-            |> List.map (fun vusb -> Db.VUSB.get_attached ~__context ~self:vusb)
-            |> List.filter (fun attached -> Db.is_valid_ref __context attached)
+            |> List.map (fun self -> Db.VUSB.get_USB_group ~__context ~self)
+            |> List.map (fun usb_group -> List.hd (Db.USB_group.get_PUSBs ~__context ~self:usb_group))
```

Comment:
We need to catch also this and raise an error saying that there is no PUSB, or we can avoid the `List.hd` and process empty strings in this case?

Reply:
yes